plugins/scaffolder-backend/src/scaffolder/stages/prepare/gitlab.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fs from 'fs-extra';
+import path from 'path';
+import os from 'os';
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import { parseLocationAnnotation } from '../helpers';
+import { InputError } from '@backstage/backend-common';
+import { PreparerBase } from './types';
+import GitUriParser from 'git-url-parse';
+import { Clone, Cred } from 'nodegit';
+import { Config } from '@backstage/config';
+
+export class GitlabPreparer implements PreparerBase {
+  private readonly privateToken: string;
+
+  constructor(config: Config) {
+    this.privateToken =
+      config.getOptionalString('catalog.processors.gitlabApi.privateToken') ??
```

Comment:
This is taking the private GitLab token from the catalogue's config. Technically, we're crossing packages here and we might want to configure the preparer token explicitly. But since we're only going to use this token to clone the repo which the catalogue already knows how to do (and is allowed to), this might be okay. What do you think?

Reply:
Hmm, this is interesting. 

I'd suggest two different tokens for now, and place it under the scaffolder config, and accept the duplication for now.

We're planning on moving to using a GithubApp for authentication for the backend plugins for Github, and I think we will be doing the same for GitLab too once we have something that works